### PR TITLE
[DOC] Add sidecar_resource_limits to pipeline spec.

### LINKED ISSUE #5001: Document the pipeline param for the worker sidecar limits
More info in #4999

### PR DESCRIPTION
**Pull request by svekars**
Closes #5001

### CODE REVIEW

File: doc/docs/master/reference/pipeline_spec.md [hunk 1]
```diff
@@ -421,6 +425,18 @@ nothing to process. For more information about scheduling GPUs see the
 [Kubernetes docs](https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/)
 on the subject.
 
+### Sidecar Resource Limits (optional)
+
+`sidecar_resource_limits` determines the upper threshold of resources
+allocated to the sidecar, init, or storage containers. You can also
```
Comment (Tryneus) [inline doc/docs/master/reference/pipeline_spec.md:431]: `sidecar_resource_limits` is only for the sidecar - the init and storage containers use `resource_limits`.

File: doc/docs/master/reference/pipeline_spec.md [hunk 2]
```diff
@@ -421,6 +425,18 @@ nothing to process. For more information about scheduling GPUs see the
...
+set the init container limit to the user container limit.
```
Comment (Tryneus) [inline doc/docs/master/reference/pipeline_spec.md:432]: Setting the init container limits to the user container limits is automatic - there is no user control over this.

File: doc/docs/master/reference/pipeline_spec.md (hunk 1 above)
Comment (svekars) [reply doc/docs/master/reference/pipeline_spec.md:431]: corrected

File: doc/docs/master/reference/pipeline_spec.md (hunk 2 above)
Comment (svekars) [reply doc/docs/master/reference/pipeline_spec.md:432]: corrected.

**Pull request merged**